Make labrad complex arrays unflatten to DimensionlessArray/ValueArray

maffoo opened the pull request:
This allows us to preserve unit information when unflattening complex
arrays, which was previously getting lost when these would be unflattened
as bare numpy ndarrays.

Fixes #64

DanielSank commented:
This looks pretty good but the organization of the test code is a bit puzzling. `test_types.py` was the place for testing flatten/unflatten operations, but the new `test_valuarray` puts those tests for ValueArray in a separate file. Is this intentional?

maffoo commented:
Good call Dan. I've added some test cases for complex ValueArray's to test_types.py, and just removed the test_valuearray file, which was redundant with the existing tests that operate on value arrays.

DanielSank commented:
Tests pass, so LGTM.
